internal/factsengine/gatherers/fstab.go

```diff
@@ -0,0 +1,100 @@
+package gatherers
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/d-tux/go-fstab"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+)
+
+const (
+	FstabGathererName = "fstab"
+	FstabFilePath     = "/etc/fstab"
+)
+
+// nolint:gochecknoglobals
+var (
+	FstabFileError = entities.FactGatheringError{
+		Type:    "fstab-file-error",
+		Message: "error reading /etc/fstab file",
+	}
+
+	FstabFileDecodingError = entities.FactGatheringError{
+		Type:    "fstab-decoding-error",
+		Message: "error decoding fstab file",
+	}
+)
+
+type FstabEntry struct {
+	Device     string   `json:"device"`
+	MountPoint string   `json:"mount_point"`
+	FS         string   `json:"fs"`
+	Options    []string `json:"options"`
+	Backup     uint8    `json:"backup"`
+	CheckOrder uint     `json:"check_order"`
+}
+
+type FstabGatherer struct {
+	fstabFilePath string
+}
+
+func NewFstabGatherer(filePath string) *FstabGatherer {
+	return &FstabGatherer{fstabFilePath: filePath}
+}
+
+func NewDefaultFstabGatherer() *FstabGatherer {
+	return &FstabGatherer{fstabFilePath: FstabFilePath}
+}
+
+func (g *FstabGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
```

Comment:
Maybe update this `g` by `f`

Reply:
g was "gatherer", but I renamed to f, no hard feelings on that :D